src/index.ts

```diff
@@ -313,15 +313,29 @@ class SubClean {
                     this.blacklist.forEach((mark: any) => {
                         let regex = null;
                         this.actions_count++;
+                        const nodeText:string = node.data.text;
 
                         if (mark.startsWith('/') && mark.endsWith('/')) {
                             // remove first and last characters
                             regex = new RegExp(mark.substring(1, mark.length - 1), 'i');
-                            if (regex.exec(node.data.text)) {
+                            if (regex.exec(nodeText)) {
                                 this.log(`[Match] Advertising found in node ${index + 1} (${mark})`);
-                                if (this.args.debug) this.log('[Line] ' + node.data.text);
+                                if (this.args.debug) this.log('[Line] ' + nodeText);
                                 hits++;
-                                node.data.text = '';
+                                if(index === 0) node.data.text = '';
+                                else{
+                                    const previousNodeText = (nodes[index-1].data as Cue).text;
```

Comment:
Is `x as Cue` necessary here?

Reply:
I had to import and use Cue as it got confused with other objects with "data" attribute, if it compiles and works as intended on your machine then you can remove it